frontend/public/components/namespace.jsx

```diff
@@ -1044,106 +1110,104 @@ const namespaceBarDropdownDispatchToProps = (dispatch) => ({
+        },
+      ]
+    : [];
 
-    let title = activeNamespace;
-    if (activeNamespace === ALL_NAMESPACES_KEY) {
-      title = allNamespacesTitle;
-    } else if (loaded && !_.has(items, title)) {
-      // If the currently active namespace is not found in the list of all namespaces, put it in anyway
-      items[title] = title;
-    }
-    const defaultActionItem = canCreateProject
-      ? [
-          {
-            actionTitle:
-              model.label === 'Project' ? t('public~Create Project') : t('public~Create Namespace'),
-            actionKey: CREATE_NEW_RESOURCE,
-          },
-        ]
-      : [];
-
-    const onChange = (newNamespace) => {
-      if (newNamespace === CREATE_NEW_RESOURCE) {
-        createProjectModal({
-          blocking: true,
-          onSubmit: (newProject) => {
-            setActiveNamespace(newProject.metadata.name);
-            removeQueryArgument('project-name');
-          },
-        });
-      } else {
-        onNamespaceChange && onNamespaceChange(newNamespace);
-        setActiveNamespace(newNamespace);
-        removeQueryArgument('project-name');
-      }
-    };
+  const items = data.map((item) => ({ title: item.metadata.name, key: item.metadata.name }));
 
-    return (
-      <div className="co-namespace-bar__items" data-test-id="namespace-bar-dropdown">
-        <Dropdown
-          disabled={disabled}
-          className="co-namespace-selector"
-          menuClassName="co-namespace-selector__menu"
-          buttonClassName="pf-m-plain"
-          canFavorite
-          items={items}
-          actionItems={defaultActionItem}
-          titlePrefix={model.label === 'Project' ? t('public~Project') : t('public~Namespace')}
-          title={title}
-          onChange={onChange}
-          selectedKey={activeNamespace || ALL_NAMESPACES_KEY}
-          autocompleteFilter={autocompleteFilter}
-          autocompletePlaceholder={
-            model.label === 'Project'
-              ? t('public~Select Project...')
-              : t('public~Select Namespace...')
-          }
-          userSettingsPrefix={NAMESPACE_USERSETTINGS_PREFIX}
-          storageKey={NAMESPACE_LOCAL_STORAGE_KEY}
-          shortCut={KEYBOARD_SHORTCUTS.focusNamespaceDropdown}
-        />
-        {children}
-      </div>
-    );
+  if (loaded && !items.find((option) => option.title === title) && title !== allNamespacesTitle) {
+    // Add current namespace if it isn't included
+    items.push({ title, key: title });
   }
-}
+
+  items.sort((a, b) => a.title.localeCompare(b.title));
+
+  if (canListNS) {
+    items.unshift({ title: allNamespacesTitle, key: ALL_NAMESPACES_KEY });
+  }
+
+  const onChange = (newNamespace) => {
+    if (newNamespace === CREATE_NEW_RESOURCE) {
+      createProjectModal({
+        blocking: true,
+        onSubmit: (newProject) => {
+          setActiveNamespace(newProject.metadata.name);
+          removeQueryArgument('project-name');
+        },
+      });
+    } else {
```

Comment:
We can remove this condition and move the logic to the create action handler in the namespace dropdown.

Reply:
Changed in next commit.